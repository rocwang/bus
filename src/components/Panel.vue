<template>
  <div :class="[$style.root, {[$style.collapsed]: isCollapsed}]">

    <div :class="$style.head">
      <div :class="$style.icon">
        <slot name="icon"/>
      </div>

      <h1 :class="$style.title">
        <small :class="$style.prefix">
          <slot name="prefix"/>
        </small>
        <slot name="title"/>
      </h1>

      <div :class="$style.buttons">
        <slot name="buttons"/>
      </div>

      <p :class="$style.subtitle">
        <slot name="subtitle"/>
      </p>
    </div>

    <div :class="$style.body">
      <slot name="body"/>
    </div>

  </div>
</template>

<script>
export default {
  name: "Panel",
  props: {
    isCollapsed: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style module>
.root {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 300px;
  color: var(--c-white);
  border-top: 10px solid var(--c-yellow);
  background-color: var(--c-ocean-blue);
  z-index: var(--z-panel);
  transition: transform 0.3s ease-out;
}

.root.collapsed {
  transform: translateY(218px);
}

.head {
  padding: 18px;
  display: grid;
  grid-template:
    "icon title    buttons" auto
    ".    subtitle .      " auto
    / auto 1fr auto;
  align-items: center;
}

.icon {
  width: 36px;
  grid-area: icon;
}

.title {
  grid-area: title;
  font-size: 2.4rem;
  padding-left: 18px;
}

.prefix {
  vertical-align: top;
  font-size: 1.2rem;
}

.prefix:not(:empty) {
  margin-right: 10px;
}

.buttons {
  grid-area: buttons;
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: center;
  transform: translateX(8px);
}

.subtitle {
  grid-area: subtitle;
  padding-left: 18px;
  font-size: 1.6rem;
}

.subtitle:not(:empty) {
  padding-top: 5px;
}

.body {
  padding: 10px 18px 18px;
}
</style>
