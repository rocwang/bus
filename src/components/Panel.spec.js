import { mount } from "@vue/test-utils";
import Panel from "./Panel";

describe("Panel.vue", () => {
  it("matches the snapshot", () => {
    const wrapper = mount(Panel);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("can be collapsed", () => {
    const wrapper = mount(Panel, {
      propsData: {
        isCollapsed: true
      }
    });
    expect(wrapper.classes()).toContain(wrapper.vm.$style.collapsed);
  });

  it("has 6 slots", () => {
    const wrapper = mount(Panel, {
      slots: {
        icon: '<p class="icon">test</p>',
        prefix: '<p class="prefix">test</p>',
        title: '<p class="title">test</p>',
        buttons: '<p class="buttons">test</p>',
        subtitle: '<p class="subtitle">test</p>',
        body: '<p class="body">test</p>'
      }
    });
    expect(wrapper.contains("p.icon")).toBe(true);
    expect(wrapper.contains("p.prefix")).toBe(true);
    expect(wrapper.contains("p.title")).toBe(true);
    expect(wrapper.contains("p.buttons")).toBe(true);
    expect(wrapper.contains("p.subtitle")).toBe(true);
    expect(wrapper.contains("p.body")).toBe(true);
  });
});
