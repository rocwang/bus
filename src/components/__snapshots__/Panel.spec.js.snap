// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel.vue matches the snapshot 1`] = `
<div
  class="root"
>
  <div
    class="head"
  >
    <div
      class="icon"
    />
     
    <h1
      class="title"
    >
      <small
        class="prefix"
      />
       
    </h1>
     
    <div
      class="buttons"
    />
     
    <p
      class="subtitle"
    />
  </div>
   
  <div
    class="body"
  />
</div>
`;
